pkg/registry/apiserverinternal/storageversion/strategy.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storageversion
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/apis/apiserverinternal"
+	"k8s.io/apiserver/pkg/apis/apiserverinternal/validation"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+)
+
+// storageVersionStrategy implements verification logic for StorageVersion.
+type storageVersionStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating StorageVersion objects.
+var Strategy = storageVersionStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+// NamespaceScoped returns false because all StorageVersion's need to be cluster scoped
+func (storageVersionStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// PrepareForCreate clears the status of an StorageVersion before creation.
+func (storageVersionStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+	sv := obj.(*apiserverinternal.StorageVersion)
+	sv.Status = apiserverinternal.StorageVersionStatus{}
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (storageVersionStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	sv := obj.(*apiserverinternal.StorageVersion)
+	sv.Status = apiserverinternal.StorageVersionStatus{}
```

Comment:
sorry, I meant to clear changes. `sv.Status = old.(*apiserverinternal.StorageVersion).Status`

Reply:
nice catch